### ISSUE TITLE
transport removal crashes posix meterp

### ISSUE THREAD

**Issue by rwhitcroft** [rapid7/metasploit-payloads#9]
Should be simple to reproduce:
```
meterpreter > transport add -t reverse_tcp -l 1.2.3.4 -p 4444
[*] Adding new transport ...
[+] Successfully added reverse_tcp transport.
meterpreter > transport remove -t reverse_tcp -l 1.2.3.4 -p 4444
[*] Removing transport ...

[*] 192.168.100.234 - Meterpreter session 1 closed.  Reason: Died
```

This works fine on Windows fwiw.

**Comment by OJ** [comment]
I'll fix this as part of #6.

**Comment by OJ** [comment]
Fixed in #7.
